Add the option '--if-not-exists' in the command 'ongr:es:index:create'

ndinh215 opened the pull request:
I fixed the issue #423 . Please review again. Thanks.

murnieza commented on Command/IndexCreateCommand.php:
```diff
@@ -38,7 +38,13 @@ protected function configure()
+            ->addOption('no-mapping', 'nm', InputOption::VALUE_NONE, 'Do not include mapping')
+            ->addOption(
+                'if-not-exists',
+                'ine',
```
I think that Symfony does not support multi-letter input option shortcuts. I see that we have the same with other options

ndinh215 replied:
http://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html
With this standard, it could understand wrong the options.

saimaz commented on Command/IndexCreateCommand.php:
```diff
@@ -55,6 +61,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
+                    $input->getOption('manager')
+                )
+            );
+            return;
```
I'm not sure 100% but probably better would be change to: `return 1;`.

ndinh215 replied:
You are correct. As Doctrine's code about this processing, it will be 
```
return $error ? 1 : 0;
```
whatever success or error.

saimaz commented on Tests/Functional/Command/CreateIndexCommandTest.php:
```diff
@@ -64,6 +64,37 @@ public function testExecute($managerName, $arguments, $options)
+     *
+     * @dataProvider getTestExecuteData
+     */
+    public function testExecuteWithIndexExistence($managerName, $arguments, $options)
```
You should not use different test case data provider and try to behave inside test case to match expectations. To test `--if-not-exists` please write separate test case, the data provider is not necessary.

saimaz commented on Tests/Functional/Command/CreateIndexCommandTest.php:
```diff
@@ -64,6 +64,37 @@ public function testExecute($managerName, $arguments, $options)
+        }
+
+        try {
+            $arguments['--if-not-exists'] = null;
```
it's actually an option, not argument.

saimaz commented on Tests/Functional/Command/CreateIndexCommandTest.php:
```diff
@@ -64,6 +64,37 @@ public function testExecute($managerName, $arguments, $options)
+            $manager->createIndex();
+        }
+
+        try {
```
When you will write test case, we do not need exception catch here.

ndinh215 commented:
Thanks for your points, @saimaz . With this patch, I removed ```try catch``` and added asserts for testing result which is returned from the execution and the output string. And also, I refactored the code a bit for the purpose of retrieving ```command tester``` via the method:
```
protected function getCommandTester($commandName)
    {
        $app = new Application();
        $app->add($this->getCreateCommand());

        $command = $app->find($commandName);
        $commandTester = new CommandTester($command);

        return $commandTester;
    }
```
About the data provider, I think this testcase may still use the same data provider. It could iterate all execute data and test this case with the option 'if-not-exists'.

mvar commented on Tests/Functional/Command/CreateIndexCommandTest.php:
```diff
@@ -64,6 +64,49 @@ public function testExecute($managerName, $arguments, $options)
+
+        // Try to create index in prior
+        if (!$manager->indexExists()) {
+            $manager->createIndex();
```
Index is automatically created when you call `$this->getManager()`.

mvar commented on Tests/Functional/Command/CreateIndexCommandTest.php:
```diff
@@ -64,6 +64,49 @@ public function testExecute($managerName, $arguments, $options)
+        // Test if the command output matches the expected output or not
+        $this->assertStringMatchesFormat($expectedOutput . '%a', $commandTester->getDisplay());
+
+        $manager->dropIndex();
```
Created indexes are automatically removed after each test case.

ndinh215 commented:
Closes #423

ndinh215 commented:
I will create new PR with better form for this issue.
